diff --git a/CMakeLists.txt b/CMakeLists.txt
index f3e69b845..f50304eac 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -59,6 +59,7 @@ include(CheckCXXCompilerFlag)
 include(GNUInstallDirs)
 include(CMakeDependentOption)
 
+if(0)
 set(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
 
 
@@ -419,6 +420,7 @@ endif()
 set(EIGEN_CUDA_COMPUTE_ARCH 30 CACHE STRING "The CUDA compute architecture level to target when compiling CUDA code")
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR})
+endif()
 
 # Backward compatibility support for EIGEN_INCLUDE_INSTALL_DIR
 if(EIGEN_INCLUDE_INSTALL_DIR)
@@ -475,7 +477,7 @@ endif()
 
 install(DIRECTORY Eigen DESTINATION ${INCLUDE_INSTALL_DIR} COMPONENT Devel)
 
-
+if(0)
 option(EIGEN_BUILD_DOC "Enable creation of Eigen documentation" ON)
 if(EIGEN_BUILD_DOC)
   add_subdirectory(doc EXCLUDE_FROM_ALL)
@@ -532,9 +534,10 @@ if(EIGEN_TEST_SYCL)
     add_definitions(-DEIGEN_DONT_VECTORIZE_SYCL=1)
   endif()
 endif()
+endif()
 
 add_subdirectory(unsupported)
-
+if(0)
 add_subdirectory(demos EXCLUDE_FROM_ALL)
 
 # must be after test and unsupported, for configuring buildtests.in
@@ -554,11 +557,12 @@ configure_file(scripts/cdashtesting.cmake.in cdashtesting.cmake @ONLY)
 if(BUILD_TESTING)
   ei_testing_print_summary()
 endif()
-
+endif()
 message(STATUS "")
 message(STATUS "Configured Eigen ${EIGEN_VERSION_NUMBER}")
 message(STATUS "")
 
+if(0)
 string(TOLOWER "${CMAKE_GENERATOR}" cmake_generator_tolower)
 if(cmake_generator_tolower MATCHES "makefile")
   message(STATUS "Available targets (use: make TARGET):")
@@ -586,7 +590,7 @@ message(STATUS "blas     | Build BLAS library (not the same thing as Eigen)")
 message(STATUS "uninstall| Remove files installed by the install target")
 message(STATUS "---------+--------------------------------------------------------------")
 message(STATUS "")
-
+endif()
 
 set ( EIGEN_VERSION_STRING ${EIGEN_VERSION_NUMBER} )
 set ( EIGEN_VERSION_MAJOR  ${EIGEN_WORLD_VERSION} )
